godot-core/src/builtin/vectors/vector_macros.rs

```diff
@@ -457,6 +457,42 @@ macro_rules! impl_vector_fns {
     }
 }
 
+/// Implements functions that are present only on integer vectors.
+macro_rules! impl_integer_vector_fns {
+    (
+        // Names of the components, for example `x, y`.
+        $($comp:ident),*
+    ) => {
+        /// A new vector with each component snapped to the closest multiple of the corresponding
+        /// component in `step`.
+        pub fn snapped(self, step: Self) -> Self {
+            #[inline]
```

Comment:
`#[inline]` is mostly needed for cross-crate inlining (because Rust simply can't do it without that attribute), but I wouldn't use it for private/crate-local functions. The compiler should be able to judge whether to inline such functions itself.

Reply:
Additionally, the `snap_one` method is now duplicated for all vectors. I think it would be better to declare it once manually outside the macro.

Reply:
> `#[inline]` is mostly needed for cross-crate inlining (because Rust simply can't do it without that attribute)

Not entirely true anymore, rust will automatically inline small functions as of 1.75 https://github.com/rust-lang/rust/pull/116505

Reply:
Yeah, this is however hardly useful in practice -- even if a function is "simple", there's a chance that refactoring makes it non-simple and causes pessimizations. As such, always writing `#[inline]` when inlining is allowed is still the better practice...

But it's a start, I hope this extended in the not too distant future 🙂